docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# Contribution guide 
+
+> [!NOTE]  
+> This is a project for learning! Don’t worry about making mistakes—everything here is reversible. This project is designed as a safe space to learn and experiment.
+
+## Table of Contents
+1. [Requirements](#requirements)  
+2. [Installing git](#installing-git)  
+3. [Fork the repository](#fork-the-repository)  
+4. [Choosing an issue to work on](#choosing-an-issue-to-work-on)  
+5. [Pushing changes](#pushing-changes)  
+6. [Creating a Pull Request](#creating-a-pull-request)  
+7. [Note on Code Review](#note-on-code-review)  
+   7.1 [For reviewers](#for-reviewers)  
+8. [Creating issues](#creating-issues)  
+
+## 1. Requirements
+- git
+
+## 2. Installing git
+Before cloning your forked repository to your local machine, you must have Git installed. You can find instructions for installing Git for your operating system [here](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
+
+Also, consider watching this tutorial if you haven't learned about git yet (Recommended by Leon but not required): https://www.youtube.com/watch?v=SWYqp7iY_Tc
+
+## 3. Fork the repository
+You can fork the repository by clicking <a href="https://github.com/patdel0/SAGP/fork">Fork</a>.  
+A fork is a copy of the repository that will be placed on your GitHub account.
+
+**Note:** It should create a URL that looks like the following -> `https://github.com/<your_GitHub_user_name>/SAGP`.  
+
+**For example:** `https://github.com/octocat/SAGP`.
+
+**Be Aware:** What you have created is a forked copy in a remote version on GitHub. It is not on your local machine yet.
```

Comment:
For newer developers and for convenience, add a link on how to clone a repository. The official docs should be sufficient: https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository

Reply:
Nice one! I'll add a section for that

Reply:
Thanks for the suggestion. I added it under commit [4a0fc8a](https://github.com/patdel0/SAGP/pull/11/commits/4a0fc8a0b27dea617f4470f408a3e731398c9d7d).